### PR TITLE
Carousel: add stretchSlideHeight prop option

### PR DESCRIPTION
**Pull request by ryan-hunt-priceline**

### CODE REVIEW

File: packages/carousel/src/StretchSlide/StretchSlide.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React, { useEffect, useRef } from 'react'
+import PropTypes from 'prop-types'
+
+const StretchSlide = ({ slideContents, height, setHeight }) => {
+  const ref = useRef(null)
+
+  useEffect(() => {
+    ref.current.clientHeight > height && setHeight(ref.current.clientHeight)
+  })
```
Comment (craigpalermo) [inline packages/carousel/src/StretchSlide/StretchSlide.js:9]: I think you should add `height` to the dependency array for this `useEffect`

Comment (ryan-hunt-priceline) [reply packages/carousel/src/StretchSlide/StretchSlide.js]: Added `height` and `setHeight` to the dependency array

File: packages/carousel/src/Carousel.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ import { Dots } from './Dots'
 import { ArrowButton } from './ArrowButton'
 import { Slider } from './Slider'
 import { getSlideKey, getVisibleSlidesArray, useResponsiveVisibleSlides } from './helpers'
+import { StretchSlide } from './StretchSlide/StretchSlide'
```
Comment (craigpalermo) [inline packages/carousel/src/Carousel.js:11]: n-b mind adding an `index` file so you can omit `/StretchSlide`?

**Review by craigpalermo — APPROVED** [review]

File: packages/carousel/src/Carousel.js (hunk 2 above)
Comment (ryan-hunt-priceline) [reply packages/carousel/src/Carousel.js]: Updated

**Review by sdalonzo — APPROVED** [review]

**Pull request merged**